conans/client/toolchain/cmake.py

```diff
@@ -106,46 +94,105 @@ class CMakeToolchain(object):
 
         # We are going to adjust automagically many things as requested by Conan
         #   these are the things done by 'conan_basic_setup()'
-         # To support the cmake_find_package generators:
-        {% if cmake_module_path %}
+        set(CMAKE_EXPORT_NO_PACKAGE_REGISTRY ON)
+
+        # To support the cmake_find_package generators
+        {% if cmake_module_path -%}
         set(CMAKE_MODULE_PATH {{ cmake_module_path }} ${CMAKE_MODULE_PATH})
-        {% endif%}
-        {% if cmake_prefix_path %}
+        {%- endif %}
+        {% if cmake_prefix_path -%}
         set(CMAKE_PREFIX_PATH {{ cmake_prefix_path }} ${CMAKE_PREFIX_PATH})
-        {% endif%}
+        {%- endif %}
+
+        # shared libs
+        {% if shared_libs -%}
+        message(STATUS "Conan toolchain: Setting BUILD_SHARED_LIBS= {{ shared_libs }}")
+        set(BUILD_SHARED_LIBS {{ shared_libs }})
+        {%- endif %}
 
-        {% if fpic %}
+        # fPIC
+        {% if fpic -%}
         message(STATUS "Conan toolchain: Setting CMAKE_POSITION_INDEPENDENT_CODE=ON (options.fPIC)")
         set(CMAKE_POSITION_INDEPENDENT_CODE ON)
-        {% endif %}
+        {%- endif %}
 
-        {% if set_rpath %}conan_set_rpath(){% endif %}
-        {% if set_std %}conan_set_std(){% endif %}
+        # SKIP_RPATH
+        {% if skip_rpath -%}
+        set(CMAKE_SKIP_RPATH 1 CACHE BOOL "rpaths" FORCE)
+        # Policy CMP0068
+        # We want the old behavior, in CMake >= 3.9 CMAKE_SKIP_RPATH won't affect install_name in OSX
+        set(CMAKE_INSTALL_NAME_DIR "")
+        {% endif -%}
+
+        # Parallel builds
+        {% if parallel -%}
+        set(CONAN_CXX_FLAGS "${CONAN_CXX_FLAGS} {{ parallel }}")
+        set(CONAN_C_FLAGS "${CONAN_C_FLAGS} {{ parallel }}")
+        {%- endif %}
+
+        # Architecture
+        {% if architecture -%}
+        set(CONAN_CXX_FLAGS "${CONAN_CXX_FLAGS} {{ architecture }}")
+        set(CONAN_C_FLAGS "${CONAN_C_FLAGS} {{ architecture }}")
+        set(CONAN_SHARED_LINKER_FLAGS "${CONAN_SHARED_LINKER_FLAGS} {{ architecture }}")
+        set(CONAN_EXE_LINKER_FLAGS "${CONAN_EXE_LINKER_FLAGS} {{ architecture }}")
+        {%- endif %}
 
         # C++ Standard Library
-        {%- if set_libcxx %}
+        {% if set_libcxx -%}
         set(CONAN_CXX_FLAGS "${CONAN_CXX_FLAGS} {{ set_libcxx }}")
```

Comment:
Yes, you are right, the goal is to have a very easy to understand generated ``conan_toolchain.cmake``. But there is also a balance, and I think that having a clear mapping from this file to the recipe ``CMakeToolchain`` is also very important. If this toolchain is going to have a ``.preprocessor_definitions`` and another one that is ``.libcxx``, I think having them in different sections of the ``conan_toolchain.cmake`` is very good. It lets you change one value in the recipe and check immediately in the generated file. Both will be preprocessor definitions (``D_GLIBCXX_USE_CXX11_ABI``), but not added all together, but the second one will have its own section with its own ``#C++ standard library`` comment, and it would be great to make it self-documenting and explicitly mention the future ``libcxx`` member of the ``CMakeToolchain`` class, that will let users control this value. Wdyt?